src/java.security.jgss/share/classes/sun/security/krb5/internal/HostAddress.java

```diff
@@ -78,41 +78,31 @@ public Object clone() {
     }
 
 
+    @Override
     public int hashCode() {
-        if (hashCode == 0) {
-            int result = 17;
-            result = 37*result + addrType;
-            if (address != null) {
-                for (int i=0; i < address.length; i++)  {
-                    result = 37*result + address[i];
-                }
-            }
-            hashCode = result;
+        int h = hashCode;
+        if (h == 0) {
+            hashCode = h = (37 * addrType + Arrays.hashCode(address));
         }
-        return hashCode;
-
+        return h;
     }
 
+    @Override
     public boolean equals(Object obj) {
         if (this == obj) {
             return true;
         }
 
-        if (!(obj instanceof HostAddress)) {
+        if (!(obj instanceof HostAddress h)) {
             return false;
         }
 
-        HostAddress h = (HostAddress)obj;
         if (addrType != h.addrType ||
             (address != null && h.address == null) ||
             (address == null && h.address != null))
             return false;
         if (address != null && h.address != null) {
-            if (address.length != h.address.length)
-                return false;
-            for (int i = 0; i < address.length; i++)
-                if (address[i] != h.address[i])
-                    return false;
+            return Arrays.equals(address, h.address);
```

Comment:
Can't this block be combined into:
return` addrType == h.addrType && Arrays.equals(address, h.address); `?

Reply:
Have a look at the end result (commits e71fc42a173 and d9316270976), is it better?

Reply:
Oops: forgot to commit and push 608258e.